internal/nexodus/ctlconnectivty.go

```diff
@@ -0,0 +1,104 @@
+package nexodus
+
+import (
+	"encoding/json"
+	"fmt"
+	"go.uber.org/zap"
+	"net"
+)
+
+const batchSize = 10
+
+func (ac *NexdCtl) Connectivity(_ string, keepaliveResults *string) error {
+	res := ac.ax.connectivityProbe()
+	var err error
+
+	// Marshal the map into a JSON string.
+	keepaliveJson, err := json.Marshal(res)
+	if err != nil {
+		return fmt.Errorf("error marshalling connectivty results")
+	}
+
+	*keepaliveResults = string(keepaliveJson)
+
+	return nil
+}
+
+func (ax *Nexodus) connectivityProbe() map[string]KeepaliveStatus {
```

Comment:
Double check me here plx, I'm guilty of not testing with user space but I think this will cover it:

```go
	if ax.userspaceMode {
		// TODO: to be addressed in #703 'nexd proxy: Add support for nexd keepalives'
		ax.logger.Warn("nexctl connection checks are currently unsupported for nexd user space proxy")
		return make(map[string]KeepaliveStatus)
	}
```